### PR TITLE
Cleanup DC_PlaceDataInHardwareDataWave

### PR DESCRIPTION
**Pull request by t-b**

### CODE REVIEW

**Comment by MichaelHuth** [comment]
The usage of `testPulseAmplitude` and `DAScale `should also be simplified.
I think the original intention was to copy all scaling in line 819 ff. to DAScale and use only DAScale when the actual DA channels are setup.
However currently also testPulseAmplitude is used there. 

Also in the first inner if testPulseAmplitude is not set, which might be a bug, as my guess is that if DAScale is set to zero then testPulseAmplitude should also be zero?

How about the following simplification:

- We keep TestPulseAmplitude and DAScale split for TP and DAQ instead of mixing both in DAScale.

- We introduce a local ChannelEnable wave that contains either 1 or 0 per channel and is multiplied on DA setup to the DA data. It is set in the line 819 ff.

**Comment by timjarsky** [comment]
> Also in the first inner if testPulseAmplitude is not set, which might be a bug, as my guess is that if DAScale is set to zero then testPulseAmplitude should also be zero?
> 

The TP (insert) settings should be independent of the DA scale factor.

The current behavior (see below) is correct.

image

**Comment by t-b** [comment]
@MichaelHuth Regarding https://github.com/AllenInstitute/MIES/pull/180#issuecomment-498639115. Let's use the simplification we talked about on the phone. One wave for the DAScale and the testpulseamplitude with dimension labels.

**Comment by t-b** [comment]
Note to self: Needs proper conflict resolution.

**Comment by t-b** [comment]
@MichaelHuth Can you rebase this PR and fix the conflicts?

**Comment by MichaelHuth** [comment]
@t-b  Could you send me the test results  please?

**Comment by MichaelHuth** [comment]
@t-b  Hmm locally it works, so I still need the CI test results,

**Comment by t-b** [comment]
I'll send them once I get back. The VPN has issues here.

**Comment by t-b** [comment]
AFT9 fails as it is flaky. [JU_HardwareTests-NI_2019-09-21_14-00-47.zip](https://github.com/AllenInstitute/MIES/files/3641487/JU_HardwareTests-NI_2019-09-21_14-00-47.zip)

**Comment by t-b** [comment]
@MichaelHuth 

Some tests fail:
[JU_HardwareTests-NI_2019-10-04_10-19-35.zip](https://github.com/AllenInstitute/MIES/files/3696022/JU_HardwareTests-NI_2019-10-04_10-19-35.zip)

**Comment by t-b** [comment]
@MichaelHuth I've reviewed the changes. Looks nice and is a real improvement. I had a couple of other minor fixes added as well. Can you review? And in case it is good, please rebase and force push.

File: Packages/MIES/MIES_DataConfiguratorITC.ipf [hunk 1]
```diff
@@ -750,14 +750,15 @@ static Function DC_PlaceDataInHardwareDataWave(panelTitle, numActiveChannels, da
 	variable numActiveChannels, dataAcqOrTP, multiDevice
 
 	variable i, j
-	variable activeColumn, numEntries, setChecksum, stimsetCycleID, fingerprint, hardwareType, maxITI
+	variable numDACEntries, numADCEntries, ttlIndex, setChecksum, stimsetCycleID, fingerprint, hardwareType, maxITI
```
Comment (t-b) [inline Packages/MIES/MIES_DataConfiguratorITC.ipf]: Good!

**Comment by t-b** [comment]
@timjarsky Ready when CI passes.

**Comment by timjarsky** [comment]
@t-b I broke it. It may be unique to this PR as my efforts to replicate in master failed. 

HS1, HS3 active
AD4-7 on
TTL 1 and 3

image

  !!! Assertion FAILED !!!
  Message: "Unknown clamp mode"
  Please provide the following information if you contact the MIES developers:
  ################################
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  Stacktrace:
DAP_ButtonProc_TPDAQ(...)#L3921 [MIES_DAEphys.ipf]
DQM_StartDAQMultiDevice(...)#L175 [MIES_DataAcquisition_Multi.ipf]
DC_ConfigureDataForITC(...)#L88 [MIES_DataConfiguratorITC.ipf]
DC_PlaceDataInHardwareDataWave(...)#L878 [MIES_DataConfiguratorITC.ipf]
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  Time: 2019-10-12T14:16:45-07:00
  Locked device: [ITC1600_Dev_1]
  Current sweep: [18]
  DAQ: [DAQ_BG_MULTI_DEVICE]
  Testpulse: [TEST_PULSE_NOT_RUNNING]
  Experiment: 2019_10_12_140635.pxp
  Igor Pro version: 8.0.4.2 (34528)
  MIES version:
  Release_2.0_20190602-280-gcbaf470b
Date and time of last commit: 2019-10-11T12:37:12+02:00
Submodule status: 
160000 550cea2cd1e7e77cb7b7396bb9a536f805f6a889 0	Packages/IPNWB
160000 7b6d2115733523463bb9d0250cee38a41a88d8b3 0	Packages/ITCXOP2
160000 aa1858d915ee28ebf5974f59f0bc19a421e5c28e 0	Packages/ZeroMQ
160000 da79879e19e6467daa58fb075f8428ae15df2061 0	Packages/doc/doxygen-filter-ipf
160000 a097151fbc9a8a8f2e504c9e46adcb327a19c8f3 0	Packages/unit-testing
  ################################

**Comment by t-b** [comment]
@timjarsky Thanks for the test. Fixed.

**Comment by t-b** [comment]
@MichaelHuth FYI this is the fix I did:

```diff
$ git plog 6222 -n1
commit 6222b3ac09ead907712d439386d9b0ea96700dd8
Author: Thomas Braun <thomas.braun@byte-physics.de>
Date:   Sun Oct 13 13:01:52 2019 +0200

    fixup! Cleanup of DC_PlaceDataInHardwareDataWave

diff --git a/Packages/MIES/MIES_DataConfiguratorITC.ipf b/Packages/MIES/MIES_DataConfiguratorITC.ipf
index 51a32800..dc7095ad 100644
--- a/Packages/MIES/MIES_DataConfiguratorITC.ipf
+++ b/Packages/MIES/MIES_DataConfiguratorITC.ipf
@@ -870,7 +870,7 @@ static Function DC_PlaceDataInHardwareDataWave(panelTitle, numActiveChannels, da
         maxITI = max(maxITI, WB_GetITI(stimSet[i], setColumn[i]))

         if(IsFinite(headstage))
-            channelMode = ChannelClampMode[i][%DAC][%ClampMode]
+            channelMode = ChannelClampMode[channel][%DAC][%ClampMode]
             if(channelMode == V_CLAMP_MODE)
                 DACAmp[i][%TPAMP] = TPAmpVClamp
             elseif(channelMode == I_CLAMP_MODE || channelMode == I_EQUAL_ZERO_MODE)
(base)
```

**Pull request merged**